src/git-command-manager.ts

```diff
@@ -0,0 +1,399 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as fshelper from './fs-helper';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+export interface IGitCommandManager {
```

Comment:
this is a wrapper over git.exe and exposes methods that correlate 1-to-1 with the git commands we need to run.